### DISCUSSION TITLE
TabView + LazyView w/Items Binding

### DISCUSSION THREAD

**Discussion by cprice70** [xamarin/XamarinCommunityToolkit#1348] [Q&A] [answered]
I'm trying to use TabView with LazyView but with dynamic tabs using an ObservableList bound to Items.  The tabs are created but LazyViews do not get initialized when the tab is selected.

I found a sort of workaround by calling LoadViewAsync when the Tab is selected however getting a handle to the content view is kind of hacking and doesn't work constantly.  

Is this a bug?  Feature request?  Something I'm just doing wrong?

**Comment by pictos** [comment]
@cprice70 can you share some project sample? I can't see what you're trying to do

**Reply by cprice70** [reply]
@pictos I've created a sample app here
https://github.com/cprice70/TabViewSample

In the sample, tabs are created with a fixed list of strings but in our actual project, this list is dynamically created.  Content views in the sample are LazyViews with a simple label in them.  In our project, these are LazyViews w/ CollectionViews.  I made them Labels in the sample for simplicity.

**Reply by cprice70** [reply]
@pictos Any ideas?

**Reply by pictos** [reply]
@cprice70 I'll be able to take a look at it next week (:

**Reply by pictos** [reply]
@cprice70 I see now... So right now we don't support LazyView inside DataTemplate for TabView out of the box... If you want you to need to implement yourself or open an issue and send a PR to XCT (I would love if you do the last one <3)

**Comment by pictos** [comment, marked as answer]
@cprice70 I see now... So right now we don't support LazyView inside DataTemplate for TabView out of the box... If you want you to need to implement yourself or open an issue and send a PR to XCT (I would love if you do the last one <3)
